### ISSUE TITLE
Upgrade Laravel

### ISSUE THREAD

**Issue by philippejadin** [agorakit/agorakit#496]
Addressing the elephant in the room regarding security, we currently use Laravel 8 [which is not supported anymore](https://laravel.com/docs/8.x/releases#support-policy).
I don't mind upgrading even if it's a long process that brings almost nothing usefully new, I would not even mind paying a Laravel shift or two.

At some point it will be more pressing so...

**Comment by philippejadin** [comment]
The latest merge brings the complete codebase to Laravel 11, including all dependencies.

In the process I also made the switch to intervention/image v3,  v2 being unsupported (a bit painful upgrade, but it allowed me to clean the hasCover trait and remove more code related to covers and thumbnails in other places.

Test passes, and we now require PHP 8.3 (seems that the imap library need it instead of 8.2 that Laravel 11 requires, it seemed not bad to require 8.3)

I think we need lot more tests to validate big code changes like this one, especially in the more exotic checkmailbox artisan command and the notification system. But for this I'm not super qualified (like, how one does mock an imap server?)

@linc if you can check on your side that everything works smoothly, I'd be more confident pushing a new release later next week, and try it on app.agorakit.org,  and monitor closely the logs :)

**Comment by zerline** [comment]
What is missing here?

**Comment by philippejadin** [comment]
As far as I can tell the upgrade is working fine. We need more tests but
this is covered in another issue. Also check the docs if the requirements
are up to date for instance PHP version needed.

(Quickly sent from the train 🙂)

Le mar. 3 déc. 2024, 07:43, Odile Bénassy ***@***.***> a
écrit :

**Comment by linc** [comment]
> seems that the imap library need it instead of 8.2

Can you elaborate on that? I'm not opposed to moving to 8.3, but I'd like to be sure it's necessary. PHP 8.2 has [2 more years of support](https://www.php.net/supported-versions.php) so it feels early to cut it off.

**Comment by philippejadin** [comment]
You are right, I rushed this upgrade, sorry for that. I will check if an older version of IMAP can be used, working with 8.2. this will be a clean PR this time I promise 🙂

**Comment by zerline** [comment]
What exactly is the point with IMAP?

What do you want to test in #506 ?

**Comment by philippejadin** [comment]
Agorakit allows to reply by email. When you receive a notification, you can reply to it (typical use case is in a discussion). Same as the ability to reply by mail to a github issue.

The email reply is sent to a catch all mailbox. From there, the scheduled [CheckMailbox command](https://github.com/agorakit/agorakit/blob/main/app/Console/Commands/CheckMailbox.php) open the mailbox using imap, in order to process the incoming mails. Mails are then moved to folders on the mail server using imap. There are several folders created, more for debugging purposes, because handling inbound email is **very** messy. It could work using pop3, but in this case we could not be able to move the processed emails in various folders.

Ideally several uses cases involving real emails (from various mails clients and contexts) should be tested, but frankly it's not easy.

Thank you by the way for those questions, answering them give me hints at what is missing in the developer docs.

**Comment by philippejadin** [comment]
Merged #532 . I think we can close this one, until next time Laravel bring an upgrade. We'll see how to tackle this :-)
